fsspec/mapping.py

```diff
@@ -120,11 +125,20 @@ def delitems(self, keys):
 
     def _key_to_str(self, key):
         """Generate full path for the key"""
-        if isinstance(key, (tuple, list)):
-            key = str(tuple(key))
-        else:
+        if not isinstance(key, str):
+            # raise TypeError("key must be of type `str`, got `{type(key).__name__}`"
+            warnings.warn(
+                "from fsspec 2023.5 onward FSMap non-str keys will raise TypeError,"
+                f" got: {key!r} of type `{type(key).__name__}`",
+                FutureWarning,
+                stacklevel=2,
+            )
+            if isinstance(key, list):
+                key = tuple(key)
             key = str(key)
-        return self.fs._strip_protocol("/".join([self.root, key]) if self.root else key)
+        if ":" in key and "://" in key:
```

Comment:
Honestly, I wouldn't bother checking, and let the backend FS error. Since the Map has a root, this protocol string would appear midway through the final URL. 

(I would note that the first condition is redundant, covered by the second)

Reply:
I'll revert that commit too 👍